server/src/main/resources/UI_Interaction.puml

```diff
@@ -0,0 +1,54 @@
+@startuml
+participant TerraUI
+box "Leo VM"
+    participant JupyterExtension
+    participant JupyterAutoSaveExtension
+    participant Welder
+end box
+participant GCS
+
+TerraUI -> GCS ++: isLocked?
+alt No Lock or locked by current user
+    GCS -> TerraUI:
+    note left: Enable EDIT button
+else LockBy someone else
+    return
+    note left: Disable EDIT mode button
+end
+TerraUI -> Welder: POST /storageLinks
+note left: update storagelink for a given notebook(both SAFE mode and EDIT mode)
+TerraUI -> Welder: localize
+note left: localize file to corresponding location
+
+JupyterExtension -> Welder ++ #F8C471: /checkMeta (syncMode)
+alt 200 SyncMode.EDIT
+    Welder -> JupyterExtension: EDIT mode
+    JupyterExtension -> Welder: /acquireLock
+
+    alt 200 successfully acquire Lock
+        Welder -> JupyterExtension: No Lock or locked by current user
+        loop every 30 seconds?
+            JupyterExtension -> Welder ++ #F4D03F: /checkMeta
+            alt SyncStatus.LIVE
+                Welder -> JupyterExtension: file is live
+            else SyncStatus.REMOTE_NOT_FOUND
+                Welder -> JupyterExtension: remote file deleted
+                note left: Do we assume user will just want to /safeDelocalize?
+            else SyncStatus.DESYNCHRONIZED
+                return file is out of sync
+                note left: prompt user to choose 1. make a new copy 2. go to SAFE mode with current file 3. go to SAFE mode with latest remote file
+            end
+        end
+    else lockedBy someone else
+        Welder -> JupyterExtension: Locked by someone else
+        note left: prompt user to choose 1. make a new copy  2. go to safe mode with current file 3. go to SAFE mode with latest remote file
```

Comment:
The 2 vs 3 distinction is interesting, but may be too subtle to try and explain in a modal. Discussion is continuing with Marty/Adrian on what the UX should be here